File: Algorithms/Strings/substring_bf.cpp
/*
Substring Search using Brute Force Approach
Language = C++11
Time Complexity = O(nm)

Ex. Query:
Text: abacbaabcabccabcaabb
Pattern: abcaa
*/
#include <bits/stdc++.h>

using namespace std;

void search(string txt, string pat) {
    int a = pat.length();
    int b = txt.length();
    for (int i = 0; i <= (b - a) + 1; i++) {
        int x = 0;
        while (x < a) {
            if (txt[i+x] != pat[x]) {
                break;
            }
            x++;
        }
        if (x == a) {
            printf("Index = %d\n", i);
        }
    }
}

int main() {
    search("abcaaabcabccabcaabb", "abcaa");
}

File: Algorithms/Strings/substring_bf.py
# Substring Search using Brute Force Approach
# Language = Python 2.7
"""
Ex. Query:
Text: abacbaabcabccabcaabb
Pattern: abcaa
"""

def search(txt, pat):
    a = len(pat)
    b = len(txt)
    i = 0
    while (i <= (b - a) + 1):
        x = 0
        while (x < a):
            if txt[i+x] != pat[x]:
                break
            x += 1
        if (x == a):
            print "Index = %s" %i
        i += 1

if __name__ == "__main__":
    search("abacbaabcabccabcaabb", "abcaa")